pkg/eval/external_cmd.go

```diff
@@ -29,7 +30,7 @@ type ExternalCmd struct {
 }
 
 func (ExternalCmd) Kind() string {
-	return "fn"
+	return "external"
```

Comment:
I can't grok what you're saying since "loose interface" has no meaning. Furthermore, it seems to me that if I do `cmd = (external true); kind-of $cmd` it should tell me `$cmd` is an external command. Nonetheless, I'll revert this change since it isn't really the point of this PR.